src/scripts/list-users.test.ts

```diff
@@ -0,0 +1,176 @@
+    // When we attempt to list the users from the client
+    const result = await usersInGroup(
+      userService,
+      groupService,
+      test.group.id
+    )();
+
+    // Then we should have a right
+    expect(result).toBeRight();
+
+    // And if it is a right, we should get a string as a result
+    const actual = result._tag === 'Right' ? result.right : result.left;
+
+    // And we expect that the resulting string contains the ID, Email, Name, and Status of the User
+    expect(actual).toContain('user_id');
+    expect(actual).toContain('test@localhost');
+    expect(actual).toContain('test user');
+    expect(actual).toContain('ACTIVE');
+
+    // And we also expect getGroup to have been called
+    expect(groupService.getGroup).toBeCalledWith(test.group.id);
+  });
+
+  it('fails when attempting to list users within an existing group with error', async () => {
+    // Given a user service that fails to retrieve a list of users
+    const userService: UserService = {
+      ...test.baseUserService(),
+      listUsersInGroup: () => TE.left('No users.'),
+    };
+
+    // And a group service that succsessfully retreives a group
+    const groupService: GroupService = {
+      ...test.baseGroupService(),
+      // eslint-disable-next-line unused-imports/no-unused-vars-ts, @typescript-eslint/no-unused-vars
+      getGroup: jest.fn((_group) => TE.right(O.some(test.group))),
+    };
+
+    // When we attempt to list the users from the client
+    const result = await usersInGroup(
+      userService,
+      groupService,
+      test.group.id
+    )();
+
+    // Then we should have a left
+    expect(result).toEqualLeft('No users.');
+
+    // But getGroup was called
+    expect(groupService.getGroup).toBeCalledWith(test.group.id);
+  });
+
+  it('fails when attempting to list users from a non-existent group', async () => {
+    // Given a user service that succsessfully retrieves a list of users (which should be impossible if the group doesn't exist)
+    const userService: UserService = {
+      ...test.baseUserService(),
+      // eslint-disable-next-line unused-imports/no-unused-vars-ts, @typescript-eslint/no-unused-vars
+      listUsersInGroup: jest.fn((_group) => TE.right([test.user])),
+    };
+
+    // And a group service that fails to find the group
+    const groupService: GroupService = {
+      ...test.baseGroupService(),
+      // eslint-disable-next-line unused-imports/no-unused-vars-ts, @typescript-eslint/no-unused-vars
+      getGroup: (_group) => TE.left('No group.'),
+    };
+
+    // When we attempt to list the users from the client
+    const result = await usersInGroup(
+      userService,
+      groupService,
+      test.group.id
+    )();
+
+    // Then we should have a left
+    expect(result).toEqualLeft('No group.');
+
+    // And listUsersInGroup was never called
+    expect(userService.listUsersInGroup).not.toHaveBeenCalled();
+  });
+
+  it('fails when attempting to list users from a none group', async () => {
```

Comment:
from a group that does not exist

Reply:
There is already a test titled 'fails when attempting to list users from a non-existent group'. Though in that method the getGroup() method returns a TE.left instead of a none option.

Reply:
I will swap the naming around: failed to get group or retreived non-existent group

Reply:
Altered